src/github.com/matrix-org/dendrite/clientapi/routing/filter.go

```diff
@@ -97,21 +89,21 @@ func PutFilter(
 		return httputil.LogThenError(req, err)
 	}
 
-	var filter gomatrix.Filter
+	var filter gomatrixserverlib.Filter
 
 	if reqErr := httputil.UnmarshalJSONRequest(req, &filter); reqErr != nil {
 		return *reqErr
 	}
 
-	filterArray, err := json.Marshal(filter)
-	if err != nil {
+	// Validate generates a user-friendly error
+	if err = filter.Validate(); err != nil {
```

Comment:
Does this still check if the JSON is in the correct format, as the code did before?

Reply:
What do you mean by "if the JSON is in the correct format" ?

`UnmarshalJSONRequest` already checks if the JSON is valid and that types from the Filter struct match types from the json data.
`Validate()` only checks if values inside the Filter struct match those allowed by the documentation  (ie the `event_format` field)

Reply:
Ah, right. I was just looking at the `Marshal` being taken off and thinking that was the only line checking the JSON correctness.